frontend/src/container/NewDashboard/DashboardVariablesSelection/VariableItem.tsx

```diff
@@ -36,17 +57,46 @@ function VariableItem({
 
 				const response = await query({
 					query: variableData.queryValue || '',
+					variables: existingVariables,
 				});
 
 				setIsLoading(false);
 				if (response.error) {
-					setErrorMessage(response.error);
+					let message = response.error;
+					if (response.error.includes('Syntax error:')) {
+						message =
+							'Please make sure query is valid and dependent variables are selected';
+					}
+					setErrorMessage(message);
 					return;
 				}
-				if (response.payload?.variableValues)
-					setOptionsData(
-						sortValues(response.payload?.variableValues, variableData.sort) as never,
-					);
+				if (response.payload?.variableValues) {
+					const newOptionsData = sortValues(
+						response.payload?.variableValues,
+						variableData.sort,
+					) as never;
+					const oldOptionsData = sortValues(optionsData, variableData.sort) as never;
+					if (!equalsCheck(newOptionsData, oldOptionsData)) {
```

Comment:
also not sure what we are trying to achieve can you please put some comments

Reply:
Whenever we update the variables options, if there is some variable that depends on this (by regexing `{{.var}}`) we push update for that variable too.